integration-tests/test_register.py

```diff
@@ -0,0 +1,47 @@
+import contextlib
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response=insights_client.run('--register', check=False)
+    assert response.returncode == 0,\
+        "An application returns 0 after successful registration"
+    stdout = response.stdout.decode("UTF-8")
```

Comment:
the default behaviour of `run()` for classes of tools in pytest-client-tools now is to use `universal_newlines=True`, so the `decode()` can be dropped

Reply:
removed.